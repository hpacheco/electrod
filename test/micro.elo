univ : { On#0 Off#0 _A####0 _B####0 -8 -7 -6 -5 -4 -3 -2 -1 0 1 2 3 4 5 6 7 };

const _A### :1 { ( _A####0 ) };
const _B### :1 { ( _B####0 ) };
var this##Light#input :2 { } { ( _A####0 On#0 ) ( _A####0 Off#0 ) ( _B####0 On#0 ) ( _B####0 Off#0 ) };
var this##Light#state :2 { } { ( _A####0 On#0 ) ( _A####0 Off#0 ) ( _B####0 On#0 ) ( _B####0 Off#0 ) };
const Int##min :1 { ( -8 ) };
const Int##zero :1 { ( 0 ) };
const Int##max :1 { ( 7 ) };
const Int##next :2 { ( -8 -7 ) ( -7 -6 ) ( -6 -5 ) ( -5 -4 ) ( -4 -3 ) ( -3 -2 ) ( -2 -1 ) ( -1 0 ) ( 0 1 ) ( 1 2 ) ( 2 3 ) ( 3 4 ) ( 4 5 ) ( 5 6 ) ( 6 7 ) };
const seq##Int :1 { ( 0 ) ( 1 ) ( 2 ) ( 3 ) };
const String :1 { };
const this##On :1 { ( On#0 ) };
const this##Off :1 { ( Off#0 ) };
const ints :1 { ( -8 ) ( -7 ) ( -6 ) ( -5 ) ( -4 ) ( -3 ) ( -2 ) ( -1 ) ( 0 ) ( 1 ) ( 2 ) ( 3 ) ( 4 ) ( 5 ) ( 6 ) ( 7 ) };

run
(always ((_A### . this##Light#input) = (_B### . this##Light#input)) or 
 eventually (!((_A### . this##Light#state) = (_B### . this##Light#state))));
true;
