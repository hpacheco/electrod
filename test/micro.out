*** This is nuXmv 2.0.0 (compiled on Mon Oct 14 17:41:52 2019)
*** Copyright (c) 2014-2019, Fondazione Bruno Kessler
*** For more information on nuXmv see https://nuxmv.fbk.eu
*** or email to <nuxmv@list.fbk.eu>.
*** Please report bugs at https://nuxmv.fbk.eu/bugs
*** (click on "Login Anonymously" to access)
*** Alternatively write to <nuxmv@list.fbk.eu>.

*** This version of nuXmv is linked to NuSMV 2.6.0.
*** For more information on NuSMV see <http://nusmv.fbk.eu>
*** or email to <nusmv-users@list.fbk.eu>.
*** Copyright (C) 2010-2019, Fondazione Bruno Kessler

*** This version of nuXmv is linked to the CUDD library version 2.4.1
*** Copyright (c) 1995-2004, Regents of the University of Colorado

*** This version of nuXmv is linked to the MiniSat SAT solver. 
*** See http://minisat.se/MiniSat.html
*** Copyright (c) 2003-2006, Niklas Een, Niklas Sorensson
*** Copyright (c) 2007-2010, Niklas Sorensson

*** This version of nuXmv is linked to MathSAT
*** Copyright (C) 2009-2019 by Fondazione Bruno Kessler
*** Copyright (C) 2009-2019 by University of Trento and others
*** See http://mathsat.fbk.eu

-- no proof or counterexample found with bound 4
-- LTL specification !( G ((J-c-a <-> J-d-a) & (J-c-b <-> J-d-b)) |  F !((I-c-a <-> I-d-a) & (I-c-b <-> I-d-b)))  is false
-- as demonstrated by the following execution sequence
Trace Description: IC3 counterexample 
Trace Type: Counterexample 
  -> State: 1.1 <-
    J-d-a = FALSE
    J-c-a = FALSE
    J-d-b = FALSE
    J-c-b = TRUE
    I-d-a = FALSE
    I-c-a = FALSE
    I-d-b = FALSE
    I-c-b = TRUE
  -- Loop starts here
  -> State: 1.2 <-
    J-d-a = FALSE
    J-c-a = FALSE
    J-d-b = FALSE
    J-c-b = FALSE
    I-d-a = FALSE
    I-c-a = FALSE
    I-d-b = FALSE
    I-c-b = FALSE
  -> State: 1.3 <-
    J-d-a = FALSE
    J-c-a = FALSE
    J-d-b = FALSE
    J-c-b = FALSE
    I-d-a = FALSE
    I-c-a = TRUE
    I-d-b = FALSE
    I-c-b = FALSE
  -> State: 1.4 <-
    J-d-a = FALSE
    J-c-a = FALSE
    J-d-b = FALSE
    J-c-b = FALSE
    I-d-a = FALSE
    I-c-a = FALSE
    I-d-b = FALSE
    I-c-b = FALSE

